extensions/cl_ext_buffer_device_address.asciidoc

```diff
@@ -0,0 +1,289 @@
+----
+
+== Modifications to the OpenCL API Specification
+
+=== Section 5.2.1 - Creating Buffer Objects:
+
+Add new allocation flags <<clCreateBuffer, List of supported memory flag values table>>:
+
+[[list-of-supported-memory-flag-values-adds]]
+.List of supported memory flags by {clCreateBuffer}
+[width="100%",cols="<50%,<50%",options="header"]
+|====
+| Memory Flags | Description
+| {CL_MEM_DEVICE_ADDRESS_EXT_anchor}
+
+include::{generated}/api/version-notes/CL_MEM_DEVICE_ADDRESS_EXT.asciidoc[]
+  | This flag specifies that the buffer must have a single fixed address
+   for its lifetime and the address should be unique at least across the devices
+   of the context, but not necessarily withing the host (virtual) memory.
+
+   The flag might imply that the buffer will be "pinned" permanently to
+   a device's memory, but might not be necessarily so, as long as the address
+   range of the buffer remains constant.
+
+   The address is guaranteed to remain the same until the buffer is freed, and
+   the address can be queried via {clGetMemObjectInfo}.
+
+   The device-specific buffer content updates are still performed by
+   implicit or explicit buffer migrations performed by the runtime or the
+   client code. If any of the devices in the context does not support
+   this type of allocations, an error (CL_INVALID_VALUE) is returned.
+| {CL_MEM_DEVICE_PRIVATE_EXT_anchor}
+
+include::{generated}/api/version-notes/CL_MEM_DEVICE_PRIVATE_EXT.asciidoc[]
+  | If this flag is combined with CL_MEM_DEVICE_ADDRESS_EXT, each device in
+   the context can have their own (fixed) device-side address and copy of
+   the created buffer which are synchronized implicitly by the runtime.
+   The main difference to a default cl_mem allocation in that case is then
+   that the addresses are queriable with CL_MEM_DEVICE_PTRS_EXT and the
+   per-device address is guaranteed to be the same for the entire lifetime
+   of the cl_mem.
+|====
+
+// refError
+
+=== Section 5.5.6 - Memory Object Queries
+
+Add a new information type <<clGetMemObjectInfo, List of supported param_names table>>:
+
+[width="100%",cols="<33%,<17%,<50%",options="header"]
+|====
+| Memory Object Info | Return type | Description
+| {CL_MEM_DEVICE_PTR_EXT_anchor}
+
+include::{generated}/api/version-notes/CL_MEM_DEVICE_PTR_EXT.asciidoc[]
+  | {cl_mem_device_address_EXT_TYPE}
+      | Returns the device address for a buffer allocated with
+   CL_MEM_DEVICE_ADDRESS_EXT. If the buffer was not created with the flag
+   or there are multiple devices in the context and the buffer address is
+   not the same for all of them, it returns CL_INVALID_MEM_OBJECT.
+
+| {CL_MEM_DEVICE_PTRS_EXT_anchor}
+include::{generated}/api/version-notes/CL_MEM_DEVICE_PTRS_EXT.asciidoc[]
+  | {cl_mem_device_address_pair_EXT_TYPE}
+      | Returns the device-address pairs for all devices in the context.
+      The per-device addresses might differ when the buffer was allocated
+      with the CL_MEM_DEVICE_PRIVATE_EXT enabled.
```

Comment:
What should happen if you mix devices supporting bda with those who don't? Is 0 returned for the devices without support for bda or should the entire query fail?

Reply:
Nevermind, the creation of the memory should have failed, so such a `cl_mem` object shouldn't exist.